Added test for topy

hpohekar opened the pull request:
Added basic test for `topy`.

Refactored code in `launcher.py`

hpohekar commented on tests/test_topy.py:
```diff
@@ -0,0 +1,99 @@
+
+
+def test_single_jou():
+    file_path = os.path.join(pyfluent.EXAMPLES_PATH, "jou1.jou")
```
Done. Thank you.

hpohekar commented on tests/test_session.py:
```diff
@@ -233,7 +233,13 @@ def test_create_session_from_launch_fluent_by_setting_ip_and_port_env_var(
 def test_execute_tui_commands(new_mesh_session, tmp_path=pyfluent.EXAMPLES_PATH):
     session = new_mesh_session
-    file_path = os.path.join(tmp_path, "sample_py_journal.txt")
+
```
You're welcome. @prmukherj

seanpearsonuk commented on tests/test_topy.py:
```diff
@@ -0,0 +1,136 @@
+
+
+def test_single_jou(with_launching_container):
+
```
@hpohekar are we asserting that the Python file has been created?

hpohekar replied:
@seanpearsonuk Yes, we are reading generated file contents by `returned = f.readlines()` and by using `assert returned` we are confirming that `returned` is non empty.

hpohekar replied:
@seanpearsonuk Yes, perfect catch! Now we are asserting that Python file has been generated for 1st and 2nd test.

We can not assert this for multiple journals because then it gives `invalid_argument error` for `os` as it can not read the generated file path due to concatenation of 2 file paths together.

hpohekar replied:
We are asserting that the Python file has been created for single journal name. Now we are resolving issues related to journal names as list of strings. We will update this code accordingly soon.

hpohekar replied:
We have updated the code now.

seanpearsonuk replied:
Thanks @hpohekar I was looking at a version that didn't have the Python file checks - this looks good now :)

seanpearsonuk commented on tests/test_topy.py:
```diff
@@ -0,0 +1,168 @@
+
+    assert gen_returned
+
+    while os.path.exists(gen_file_path):
```
We should always make such loops finite even when we know that it's not necessary.

hpohekar replied:
Yes, sure. Thank you.

hpohekar replied:
Done. Thank you.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -575,8 +575,12 @@ def launch_fluent(
-                        journal_name = journals.rsplit(".")[0]
-                        journal_ext = journals.rsplit(".")[1]
+                        if os.path.exists(journals):
+                            journal_name = journals.rsplit("\\")[-1].split(".")[0]
```
@hpohekar 

The "\\\\" seems like a WIndows assumption. Shall we use something designed for the job that works with any os, e.g. os.path.split()? 

Also these two lines are unnecessarily duplicated - the rhs of a single statement can just be `name, ext = `.

Also there's no need for the if/else here. It works just the same.

hpohekar replied:
Ok. We will update this as well.

seanpearsonuk replied:
(I was still editing the comment - please check all points and let me know if anything is unclear)

hpohekar replied:
> (I was still editing the comment - please check all points and let me know if anything is unclear)

Yes, sure. Thank you.

hpohekar replied:
> [Path(<path>).stem](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem) can also be used here. We probably don't need to extract the extension here.

I see. Thank you.

hpohekar replied:
Done. Thank you.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,18 +567,18 @@ def launch_fluent(
-                    journal_ext = topy.split(".")[1]
-                    launch_string += f' -i {journal_name}.{journal_ext} -command="(api-start-python-journal \\\"\\\"{journal_name}.py\\\"\\\")"'  # noqa: E501
+                    name = topy.split(".")[0]
+                    ext = topy.split(".")[1]
```
or something like:
```
name, ext = topy.split(".")[:2]
```
because of the duplication
?

hpohekar replied:
Done. Thank you.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,18 +567,18 @@ def launch_fluent(
-                        journal_ext = journals.rsplit(".")[1]
-                        name_list.append(journal_name)
-                        all_scm_journal_names += f' -i {journal_name}.{journal_ext}'
+                        name = Path(journals).stem.split('.')[0]
```
Why is _journals_ pluralised?

hpohekar replied:
Because it's list of journals.

seanpearsonuk replied:
`topy` is the list that we are iterating over. journals is one element in that list - a string, which is the name or path of a journal.

hpohekar replied:
Removed plurals. Thank you.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,18 +567,18 @@ def launch_fluent(
-                        journal_ext = journals.rsplit(".")[1]
-                        name_list.append(journal_name)
-                        all_scm_journal_names += f' -i {journal_name}.{journal_ext}'
+                        name = Path(journals).stem.split('.')[0]
```
Why are we splitting the stem?

hpohekar replied:
Removed splitting. Thank you.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,18 +567,18 @@ def launch_fluent(
-                        name_list.append(journal_name)
-                        all_scm_journal_names += f' -i {journal_name}.{journal_ext}'
+                        name = Path(journals).stem.split('.')[0]
+                        ext = journals.rsplit(".")[1]
```
Why a separate method for the extension.

Also this code should be consolidated with the earlier code. All can be done in a single way in one function..

hpohekar replied:
`Path.stem` does not give file name with extension.

seanpearsonuk commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,18 +567,18 @@ def launch_fluent(
+                    name = topy.split(".")[0]
+                    ext = topy.split(".")[1]
+                    launch_string += f' -i {name}.{ext} -command="(api-start-python-journal \\\"\\\"{name}.py\\\"\\\")"'  # noqa: E501
                 if isinstance(topy, list):
```
do you mean elif?

hpohekar replied:
Yes. Done. Thank you.

hpohekar commented:
Thank you very much @mkundu1 for refactoring this code.

mkundu1 commented on src/ansys/fluent/core/launcher/launcher.py:
```diff
@@ -567,20 +567,11 @@ def launch_fluent(
-                    all_py_journal_names += '_'.join(name_list)
-                    launch_string += f'{all_scm_journal_names} -command="(api-start-python-journal \\\"\\\"{all_py_journal_names}.py\\\"\\\")"'  # noqa: E501
+                    topy = [topy]
+                elif isinstance(topy, list):
```
This should be an if, otherwise the code won't work when the passed topy is a string.

hpohekar commented:
@mkundu1 We have updated the `topy` argument code using a function and added type check in function.

We have also tested following scenarios (Journal name and path containing space etc.) locally and all of them are working fine.

![image](https://user-images.githubusercontent.com/106588300/208610319-1ae8916f-cd49-4016-9790-bfa4f5b19534.png)
